### PR TITLE
kola: atomically update _kola_temp platform-latest symlink

### PR DESCRIPTION
**Pull request by bgilbert**
Avoid races when multiple kola runs are started in close succession.

cc @ajeddeloh

### CODE REVIEW

File: kola/harness.go [hunk 1]
```diff
@@ -460,19 +460,21 @@ func SetupOutputDir(outputDir, platform string) (string, error) {
...
-			if err := os.Remove(linkPath); err != nil {
-				return "", err
-			}
-		} else if !os.IsNotExist(err) {
+		if err == nil && (st.Mode()&os.ModeType) != os.ModeSymlink {
+			return "", fmt.Errorf("%v exists and is not a symlink", linkPath)
+		} else if err != nil && !os.IsNotExist(err) {
+			return "", err
+		}
+		if err := os.Symlink(defaultDirName, tempLinkPath); err != nil {
```
Comment (arithx) [inline kola/harness.go]: Currently `tempLinkPath` is only removed if the Rename succeeds, is this intended or should we move it to a defer'd cleanup?

Comment (bgilbert) [reply kola/harness.go]: It's being removed only if the rename fails.  If it succeeds, the link is now at its final location so we don't need to clean it up.

Comment (arithx) [reply kola/harness.go]: Whoops, brain not working this morning need more caffeine

File: kola/harness.go [hunk 2]
```diff
@@ -460,19 +460,21 @@ func SetupOutputDir(outputDir, platform string) (string, error) {
 	}
 
 	if defaulted {
+		tempLinkPath := filepath.Join(outputDir, "latest")
 		linkPath := filepath.Join(defaultBaseDirName, platform+"-latest")
+		// safety check
```
Comment (ajeddeloh) [inline kola/harness.go]: Maybe something more descriptive like `do not clobber existing files that are not symlinks` ?

**Pull request merged**